graphql/schema/types/waterfall.graphql

```diff
@@ -0,0 +1,30 @@
+input WaterfallOptions {
+  limit: Int = 7
+  projectIdentifier: String! @requireProjectAccess(permission: TASKS, access: VIEW)
+  requesters: [String!]
+}
+
+type WaterfallTask {
+  id: String!
+  status: String!
+  displayName: String!
+}
```

Comment:
To improve performance, the pipeline only `$project`s a few task fields as part of the results. I'm not even positive a `Task` would work correctly, but even so I think it's preferable to be clear that only select fields are available.

Reply:
Ahh this makes sense. My only concern is if we start adding fields to support future functionality and we end up with 2 Task structs. One other minor concern is that we can't leverage the apollo cache to keep the tasks in sync.

Reply:
Yeah, agree that is a slight risk but I think any additions to the waterfall will have to be made deliberately since they could significantly affect performance when scaled x1000s. On the other hand, having 2 structs is perhaps preferable in a sense to keep `WaterfallTask` as lean as possible regardless of any fields added to `Task`. And I think if your second concern perceptibly affected syncing between e.g. the waterfall and task page, we might be able to introduce some more bespoke cache updating